src/sys/signal.rs

```diff
@@ -86,40 +166,33 @@ impl SigSet {
         SigSet { sigset: sigset }
     }
 
-    pub fn add(&mut self, signum: SigNum) -> Result<()> {
-        let res = unsafe { libc::sigaddset(&mut self.sigset as *mut libc::sigset_t, signum) };
-
-        Errno::result(res).map(drop)
+    pub fn add(&mut self, signal: Signal) {
```

Comment:
I'm ok with pushing that change to a later PR. This one can just be the type changing to an enum.